src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const arrDate = date.split(fromFormat[3]);
+  const slash = toFormat[3];
+  const newFormat = [];
+  const dateMap = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateMap[fromFormat[i]] = arrDate[i];
+  }
```

Comment:
The loop iterates over `fromFormat.length - 1`, which might miss the last element if the format array is longer than expected. Consider iterating over the entire array to ensure all parts are mapped.